### PR TITLE
feat: add smart-opposite resize option

### PR DESCRIPTION
**Pull request by wesbos**
Fixes https://github.com/nikitabobko/AeroSpace/issues/741

I've never worked on a swift app, but it seems like a small add so hopefully this is OK.

### CODE REVIEW

File: Sources/Common/cmdArgs/impl/ResizeCmdArgs.swift [hunk 1]
```diff
@@ -30,7 +30,7 @@ public struct ResizeCmdArgs: CmdArgs {
     }
 
     public enum Dimension: String, CaseIterable, Equatable {
-        case width, height, smart
+        case width, height, smart, smartOpposite
```
Comment (nikitabobko) [inline Sources/Common/cmdArgs/impl/ResizeCmdArgs.swift:33]:
```suggestion
        case width, height, smart
        case smartOpposite = "smart-opposite"
```

File: Sources/AppBundleTests/command/ResizeCommandTest.swift [hunk 2]
```diff
@@ -10,6 +10,10 @@ final class ResizeCommandTest: XCTestCase {
         testParseCommandSucc("resize smart -10", ResizeCmdArgs(rawArgs: [], dimension: .smart, units: .subtract(10)))
         testParseCommandSucc("resize smart 10", ResizeCmdArgs(rawArgs: [], dimension: .smart, units: .set(10)))
 
+        testParseCommandSucc("resize smart-opposite +10", ResizeCmdArgs(rawArgs: [], dimension: .smartOpposite, units: .add(10)))
+        testParseCommandSucc("resize smart-opposite -10", ResizeCmdArgs(rawArgs: [], dimension: .smartOpposite, units: .subtract(10)))
+        testParseCommandSucc("resize smartopposite 10", ResizeCmdArgs(rawArgs: [], dimension: .smartOpposite, units: .set(10)))
```
Comment (nikitabobko) [inline Sources/AppBundleTests/command/ResizeCommandTest.swift:15]:
```suggestion
        testParseCommandSucc("resize smart-opposite 10", ResizeCmdArgs(rawArgs: [], dimension: .smartOpposite, units: .set(10)))
```

File: Sources/AppBundle/command/impl/ResizeCommand.swift [hunk 3]
```diff
@@ -31,6 +31,11 @@ struct ResizeCommand: Command { // todo cover with tests
                 node = first
                 parent = first.parent as! TilingContainer
                 orientation = parent.orientation
+            case .smartOpposite:
+                guard let first = candidates.first else { return false }
+                node = first
+                parent = first.parent as! TilingContainer
+                orientation = parent.orientation == .h ? .v : .h
```
Comment (nikitabobko) [inline Sources/AppBundle/command/impl/ResizeCommand.swift:38]:
```suggestion
                orientation = parent.orientation.opposite
```

**Review by nikitabobko — CHANGES_REQUESTED** [review]
Please ping me back when you fix the CI

**Comment by nikitabobko** [comment]
You also need to update this line https://github.com/nikitabobko/AeroSpace/pull/742/files#diff-7a3999a307e560d8f439d95603d6b511bfaefe93607e2e0494505be04af95c0cR12

**Comment by wesbos** [comment]
@nikitabobko Thanks for your help on this - have the CI passing now

**Comment by nikitabobko** [comment]
Thanks, merged with the following patch:
```diff
diff --git a/Sources/AppBundle/command/impl/ResizeCommand.swift b/Sources/AppBundle/command/impl/ResizeCommand.swift
index 0bd64498..cd7e5905 100644
--- a/Sources/AppBundle/command/impl/ResizeCommand.swift
+++ b/Sources/AppBundle/command/impl/ResizeCommand.swift
@@ -32,10 +32,11 @@ struct ResizeCommand: Command { // todo cover with tests
                 parent = first.parent as! TilingContainer
                 orientation = parent.orientation
             case .smartOpposite:
-                guard let first = candidates.first else { return false }
+                guard let _orientation = (candidates.first?.parent as? TilingContainer)?.orientation.opposite else { return false }
+                orientation = _orientation
+                guard let first = candidates.first(where: { ($0.parent as! TilingContainer).orientation == orientation }) else { return false }
                 node = first
                 parent = first.parent as! TilingContainer
-                orientation = parent.orientation.opposite
         }
         let diff: CGFloat = switch args.units.val {
             case .set(let unit): CGFloat(unit) - node.getWeight(orientation)
```

otherwise the code crashes.

Please next time always run your code before submitting it, even if it looks trivial

**Pull request closed without merging**